torch/_prims/__init__.py

```diff
@@ -1589,35 +1613,35 @@ def _rev_meta(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
 #
 
 
-def _select_meta(
+def _where_meta(
     pred: TensorLikeType, a: TensorLikeType, b: TensorLikeType
 ) -> TensorLikeType:
-    utils.check_same_device(pred, a, b, allow_cpu_scalar_tensors=True)
-    utils.check_same_shape(pred, a, b, allow_cpu_scalar_tensors=True)
-    assert pred.dtype is torch.bool
 
     return _elementwise_meta(
-        a, b, type_promotion=ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND.DEFAULT
+        a,
+        b,
+        type_promotion=ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND.DEFAULT,
+        args_with_different_dtypes=(pred,),
     )
 
 
-def _select_aten(pred: Tensor, a: Tensor, b: Tensor) -> Tensor:
+def _where_aten(pred: Tensor, a: Tensor, b: Tensor) -> Tensor:
```

Comment:
out of curiosity, why do we need this helper, and not just use `torch.where` in make_prim?

Reply:
Great point -- I was just on automatic mode -- fixed!